src/zarr/array.py

```diff
@@ -343,6 +343,16 @@ def ndim(self) -> int:
     def shape(self) -> ChunkCoords:
         return self.metadata.shape
 
+    @property
+    def chunks(self) -> ChunkCoords:
```

Comment:
Should we add a deprecation warning here? I think users should instead look at the chunk grid directly.

Reply:
Is the chunk grid object is guaranteed to have a consistent structure, beyond `{name: str, configuration: dict[str, JSON]}`? We might instead want users to look at a normalized (explicit) representation of the chunks, i.e. `tuple[tuple[int, ...], ...]`

Reply:
My opinion is that having a backwards compatible `.chunks` attribute is very important. Deprecating this or changing its behavior will have far-reaching consequences for libraries that depend on Zarr. Given that we only support regular chunk grids so far, this solution feels very reasonable to me. Once other chunk grids are developed this will raise an attribute error, and the user code can handle it some other way (i.e. looking at the chunk grid directly).

Reply:
@rabernat I think your reasoning makes sense, given that irregular chunks are prospective at this point. 

Thinking about future changes to support irregular chunks, do you think it would make sense to transform the type of `chunks` to `tuple[tuple[int, ...], ...]` (matching the `chunks` attribute of a dask array), assuming we give a long deprecation warning?

Reply:
> My opinion is that having a backwards compatible `.chunks` attribute is very important. Deprecating this or changing its behavior will have far-reaching consequences for libraries that depend on Zarr. Given that we only support regular chunk grids so far, this solution feels very reasonable to me. Once other chunk grids are developed this will raise an attribute error, and the user code can handle it some other way (i.e. looking at the chunk grid directly).

Fair enough, we can revisit this once we have other chunk grids.



> `tuple[tuple[int, ...], ...]`

How would that scale if you have billions of chunks?